### PR TITLE
Set on click

### PR DESCRIPTION
**Pull request by vuhust19081995**

### CODE REVIEW

File: app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java [hunk 1]
```diff
@@ -37,20 +42,38 @@ public int getItemCount() {
         return mMovies != null ? mMovies.size() : 0;
     }
 
+    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
+        mClickListener = onItemClickListener;
+    }
+
+    public interface OnItemClickListener {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/moviedb/screen/main/genres/GenreAdapter.java [hunk 2]
```diff
@@ -38,22 +41,42 @@ public int getItemCount() {
         return mGenres != null ? mGenres.size() : 0;
     }
 
+    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
+        mOnItemClickListener = onItemClickListener;
+    }
+
+    public interface OnItemClickListener {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/main/genres/GenreAdapter.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/moviedb/data/model/Video.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Created by workspace on 27/09/2017.
+ */
+
+public class Video extends BaseModel {
+    @SerializedName("key")
+    @Expose
+    private String keyVideo;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/data/model/Video.java]: pls check convention

File: app/src/main/java/com/framgia/moviedb/data/model/VideoResponse.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.List;
+
+/**
+ * Created by workspace on 27/09/2017.
+ */
+
+public class VideoResponse extends BaseModel {
+    @SerializedName("results")
+    @Expose
+    private List<Video> mVideoList;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/data/model/VideoResponse.java]: pls rename to mVideos

File: app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java [hunk 5]
```diff
@@ -14,7 +15,11 @@
  */
 
 public class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.MoviewViewHolder> {
+    private boolean mLoading = false;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java]: default by false

File: app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java [hunk 6]
```diff
@@ -37,20 +42,42 @@ public int getItemCount() {
...
             super(binding.getRoot());
             mBinding = binding;
+            mClickListener = onItemClickListener;
         }
 
         public void bind(Movie movie) {
+            mMovie = movie;
             mBinding.setMovie(movie);
             mBinding.executePendingBindings();
+            mBinding.getRoot().setOnClickListener(this);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java]: bind listener in layout xml

File: app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java [hunk 7]
```diff
@@ -14,7 +14,10 @@
  */
 
 public class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.MoviewViewHolder> {
+    private int mLastVisibleItem;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java]: remove unsed variable mLastVisibleItem,, mTotalItemCount

File: app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java [hunk 8]
```diff
@@ -37,19 +40,34 @@ public int getItemCount() {
...
+    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
+        mClickListener = onItemClickListener;
+    }
+
+    /**
+     * Create listener
+     */
+
+    public interface OnItemClickListener {
+        void itemClick(Movie movie);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/MovieAdapter.java]: pls change menthod to onClick

File: app/src/main/java/com/framgia/moviedb/screen/home/slidebanner/BannerFragmentViewModel.java [hunk 9]
```diff
@@ -11,12 +11,18 @@
...
 
     private String mPath;
 
-    public BannerFragmentViewModel(String path) {
+    private int mIdMovie;
+
+    public BannerFragmentViewModel(Context context, String path, int idMovie) {
+        mContext = context;
         mPath = path;
+        mIdMovie = idMovie;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/home/slidebanner/BannerFragmentViewModel.java]: pls change variable name to movieId

File: app/src/main/java/com/framgia/moviedb/screen/home/slidebanner/BannerFragmentViewModel.java [hunk 10]
```diff
@@ -39,8 +45,7 @@ public String getPath() {
         return mPath;
     }
 
-    public void setPath(String path) {
-        mPath = path;
-        notifyPropertyChanged(BR.path);
+    public void onClick() {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/home/slidebanner/BannerFragmentViewModel.java]: E làm gì với menthod này?

File: app/src/main/java/com/framgia/moviedb/screen/main/genres/GenreAdapter.java [hunk 11]
```diff
@@ -16,6 +16,8 @@
 public class GenreAdapter extends RecyclerView.Adapter<GenreAdapter.BindingHolder> {
     private List<Genre> mGenres;
 
+    private GenreAdapter.OnItemClickListener mOnItemClickListener;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/main/genres/GenreAdapter.java]: pls define a base listener for item click

File: app/src/main/java/com/framgia/moviedb/screen/home/slidebanner/BannerFragmentViewModel.java (hunk 10 above)
Comment (vuhust19081995) [reply app/src/main/java/com/framgia/moviedb/screen/home/slidebanner/BannerFragmentViewModel.java]: Em tính chuyền movie sang detail, em xin sửa lại 1 chút phần này ở pull sau ạ

File: app/src/main/java/com/framgia/moviedb/screen/OnRecyclerViewItemClickListener.java [hunk 12]
```diff
@@ -0,0 +1,9 @@
+package com.framgia.moviedb.screen;
+
+/**
+ * Created by workspace on 28/09/2017.
+ */
+
+public interface OnRecyclerViewItemClickListener<T> {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/OnRecyclerViewItemClickListener.java]: WARNING: Type Javadoc comment is missing an @param <T> tag.

File: app/src/main/res/layout/item_genre.xml [hunk 13]
```diff
@@ -9,6 +9,12 @@
             name="viewModel"
             type="com.framgia.moviedb.data.model.Genre"
             />
+        <variable
+            name="itemClickListener"
+            type="com.framgia.moviedb.screen.OnRecyclerViewItemClickListener"
+            />
+        />
+        />
```
Comment (DoanVanToan) [inline app/src/main/res/layout/item_genre.xml]: WARNING: [ExtraText] Unexpected text found in layout file: "/>
        />"
 Layout resource files should only contain elements and attributes. Any XML text content found in the file is likely accidental (and potentially dangerous if the text resembles XML and the developer believes the text to be functional)

File: app/src/main/java/com/framgia/moviedb/screen/main/genres/GenreAdapter.java [hunk 14]
```diff
@@ -38,20 +41,35 @@ public int getItemCount() {
...
     }
 
+    public void setOnItemClickListener(OnRecyclerViewItemClickListener<Genre> onItemClickListener) {
+        mOnItemClickListener = onItemClickListener;
+    }
+
+    /**
+     * Create listener
+     */
+    public interface OnItemClickListener {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/main/genres/GenreAdapter.java]: remove this listener

File: app/src/main/res/layout/item_movie.xml [hunk 15]
```diff
@@ -8,6 +8,11 @@
             name="movie"
             type="com.framgia.moviedb.data.model.Movie"
             />
+        <variable
+            name="clickListener"
```
Comment (DoanVanToan) [inline app/src/main/res/layout/item_movie.xml]: pls rename to itemClickListener

File: app/src/main/res/layout/item_genre.xml [hunk 16]
```diff
@@ -17,6 +21,7 @@
         android:layout_gravity="center"
         android:background="@color/color_black"
         android:gravity="center"
```
Comment (DoanVanToan) [inline app/src/main/res/layout/item_genre.xml]: pls rename viewModel to genre

File: app/src/main/java/com/framgia/moviedb/utils/Constant.java [hunk 17]
```diff
@@ -18,6 +18,7 @@
     public static final String DRAWER_OPEN = "OPEN";
     public static final String DRAWER_CLOSE = "CLOSE";
     public static final String BUNDLE_PATH = "path";
+    public static final String BUNDLE_ID_MOVIE = "idMovie";
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/Constant.java]: pls change BUNDLE_ID_MOVIE = "BUNDLE_ID_MOVIE "
and the same for other BUNDLE

**Pull request merged**